frontend/src/views/Lobby.tsx

```diff
@@ -51,13 +52,31 @@ function LobbyPage() {
    */
   const setStateFromRoom = (room: Room) => {
     setHost(room.host);
+    setUsers(room.users);
     setActiveUsers(room.activeUsers);
     setInactiveUsers(room.inactiveUsers);
     setRoomId(room.roomId);
     setActive(room.active);
     setDifficulty(room.difficulty);
   };
 
+  /**
+   * Rebuild the room objet from the current variables.
+   * This method assumes all variables of the Room are included, and
+   * it uses useCallback so so it is not re-built in
+   * the useEffect function.
+   */
+  const rebuildRoom = useCallback((): Room => ({
+    roomId: currentRoomId,
+    host: host!,
+    users: users!,
+    activeUsers: activeUsers!,
+    inactiveUsers: inactiveUsers!,
+    active,
+    difficulty: difficulty!,
```

Comment:
Thoughts on just sending the `roomId` and `currentUser`? I'm thinking if we did something similar to the lobby page where we just pass those two and have a GET request or something, we might be able to simplify the stuff held in location state, but I'm not sure - we can talk about it during standup

Reply:
On a related note, if we decide on reducing this to just 1-2 variables, what are your thoughts on removing this callback and just adding the logic to the useEffect block itself? Since that's the only place it's used and it would end up just being something like `const params = { roomId: currentRoomId, user: currentUser }`

Reply:
I replaced the code to just have a `roomId` and `currentUser`, and I removed this callback as well as you suggested -- these are both good ideas, thanks! I haven't added a `GET` request to get the whole room, since I think this could be changed if we get a Game object, for example, on that page, and I don't think it's necessary to add that to this PR.

I also reflected the `roomId` and `currentUser` on the frontend on the Game page.